### DISCUSSION TITLE
Quarto dashboard deployed in server (.venv)

### DISCUSSION THREAD

**Discussion by LOCEANlloydizard** [quarto-dev/quarto-cli#10950] [Q&A] [answered]
### Description

Hi everybody,

I have an app coded in Quarto/Py-Shiny. It sends queries to a database and plots some data.

The app worked fine locally (with WAMP and MySQL), so I decided to deploy it on a server.
The database was also uploaded to the server, so when I preview my app in VS Code with:

`
quarto preview /home/lizard/visu/visualise_app.qmd --no-browser --no-watch-inputs
`

it works fine. Now, when I compile the app with:

`
quarto render /home/lizard/visu/visualise_app.qmd
`

I get an error in a log file I created: "No such comm target registered: quarto_kernel_setup". When I launch the  `````visualise_app.html````` created by `quarto render` the input are there etc, but no connection to the BD is made it seems, but I have no idea whats going in the back-end; how could I have an output for error? (like when I launch the app' with the `quarto preview`, errors are shown in the terminal).

I can't figure out where this comes from. I've emptied all my app code, leaving only the YAML and a bit of code but I still get this error/message. Note that I have my module in a virtual environment using source ~/.venv/bin/activate.
I also have one warning that I do not know how to address:

```bash
"(.venv) lizard@visu-drifter-int:~$ quarto render /home/lizard/visu/visualise_app.qmd
WARN: Specified QUARTO_PYTHON 'C:\Users\lloyd\anaconda3\python.exe' does not exist
WARN: No python binary found in specified QUARTO_PYTHON location 'C:\Users\lloyd\anaconda3\python.exe'"

(then continues compiling)

```

Thank you for any feedback! 
Wishing you a great evening.

Here's my example =>

````qmd
---
title: "Processing visualisation"
author: "me"
format:
  dashboard
server: shiny
logo: ./mylogo.png
---

```{python}
import os
from shiny import render, reactive, ui
from shinywidgets import render_widget, output_widget
import plotly.graph_objects as go
import plotly.express as px
import matplotlib.pyplot as plt
import pandas as pd
import polars as pl
import connectorx
import numpy as np
import mysql.connector
from mysql.connector import Error
from datetime import datetime
import time
import logging

# Configure logging to write to a file
logging.basicConfig(filename='./app_logs.log', level=logging.INFO, format='%(asctime)s - %(message)s')

# ############# Server
uri = "mysql://root:@localhost/my_BD"
logging.info(uri)
myhost='localhost'
myuser='root'
mypwd=''
myDB='my_BD'

```

# {.sidebar}

```{python}
logging.info('hey')
color_l0 = '#E60000'
color_l1 = '#228B22'
color_l2 = '#1C1C1C'

size_l0 = 60
size_l1 = 40
size_l2 = 20

ui.input_select("Principal", "Select :", choices=['Echo', 'Charlie'])
ui.input_select("type", "Select type:", choices=[])
ui.input_select("id", "Select ID", choices=[])

logging.info('tmp')
```

````

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

Did you read https://quarto.org/docs/dashboards/deployment.html#shiny-dashboards ?

**Reply by LOCEANlloydizard** [reply]
Hi Mcanouil,

I've formatted my previous message, thank you.
After reading the page you mentioned more attentively, I think that what I'm trying to do might be impossible.

"_**Dashboards that use Shiny for Python or Shiny for R are web applications that require a server for deployment (so they can't just be deployed to a generic web host).**_" => I'm using Shiny and trying to deploy it to my server...

**Reply by mcanouil** [reply, marked as answer]
You might want to look for "shinylive" the WASM version then.

**Reply by LOCEANlloydizard** [reply]
Yes, you're right. I forgot about that, makes sense!
 Thank you!

**Reply by LOCEANlloydizard** [reply]
Hi @mcanouil 

I've tried to adapt my code following this => https://github.com/quarto-ext/shinylive/blob/main/README.md

I've made my app' code way simpler to try to make it run, but I have errors on packages: polars, logging. (see below)
Maybe there are not working with shinylive?

Also, I have in total 19 blocks that use to work with just {python} blocks, but they should not be standalones, as they refers to varaibles within each other.. is that possible with shinylive?

Maybe I could code it without quarto, directly with a R shiny app and deploy it with 

```` 
shiny create --dir myapp
shinylive export myapp site
```` 

Thank you for your feedback, I'm getting a bit lost with quarto, shiny, shinylive.. 

Anyway, the code is =>

```` qmd

---
title: "My app"
author: "me"
format: html
filters:
  - shinylive
logo: ./my_logo.png
---

```{shinylive-python}
#| standalone: true

import os

from shiny import render, reactive, ui
from shinywidgets import render_widget, output_widget

import plotly.graph_objects as go
import plotly.express as px 
import matplotlib.pyplot as plt

import pandas as pd
import polars as pl
import connectorx

import numpy as np
import mysql.connector
from mysql.connector import Error
from datetime import datetime

import time
import logging

# Configure logging to write to a file
logging.basicConfig(filename='./app_logs.log', level=logging.INFO, format='%(asctime)s - %(message)s')

logging.info("test")

############# Local
uri = "mysql://root:@localhost/myBD"
myhost='localhost'
myuser='root'
mypwd=''
myDB='myDB'

```

# {.sidebar}

```{shinylive-python}
#| standalone: true

logging.info("test")
color_l0 = '#E60000'
color_l1 = '#228B22'
color_l2 = '#1C1C1C'

size_l0 = 60
size_l1 = 40
size_l2 = 20

ui.input_select("prin_id", "Select:", choices=['1', '2', '3']
ui.input_select("prin_type", "Select type:", choices=[])
ui.input_select("second_id", "Select ID", choices=[])

```


````

Then I try to preview the app', it runs (see below) but in the app i have errors on packages: polars, logging.

`
PS C:\Users\lloyd\Desktop\SWOT_AdAC\Tools\Drifter_ToolBox\04_app_from_db> quarto preview c:/Users/lloyd/Desktop/SWOT_AdAC/Tools/Drifter_ToolBox/04_app_from_db/visualise_app.qmd --no-browser --no-watch-inputs
pandoc 
  to: html
  output-file: visualise_app.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Drifter processing visualisation
  author: SWOT-AdAC
  logo: ./swot_adac_logo.png
  
Output created: visualise_app.html

Watching files for changes
Browse at http://localhost:3887/visualise_app.html 
`

The structure of the app'

````bash
my_path\04_APP_FROM_DB
├───.quarto
│   └───xref
├───visualise_app_files
│   ├───figure-html
│   ├───libs
│   │   ├───bootstrap
│   │   ├───clipboard
│   │   ├───quarto-dashboard
│   │   └───quarto-html
│   └───mediabag
└───_extensions
    └───quarto-ext
        └───shinylive
            └───resources
                └───css
````

**Reply by mcanouil** [reply]
I'm not a big user of Shiny in general and did not try Shiny for Python so far.

I would suggest to read Shiny's documentation: https://shiny.posit.co/py/

**Reply by LOCEANlloydizard** [reply]
Hi @mcanouil,

I've marked your previous message as the answer, as I think using "shinylive" should have fixed my problem.
I say "should" because it did not; I got lost in Quarto/Shiny-Python/ShinyLive, etc., so I ended up coding the app with R Shiny.

I will definitely give Quarto and Shiny-Python another try in the near future.

Thank you!

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

Did you read <https://quarto.org/docs/dashboards/deployment.html#shiny-dashboards>?
